pkg/resource/fake/mocks.go

```diff
@@ -204,6 +204,19 @@ func (m *CompositionUpdater) GetCompositionUpdatePolicy() *xpv1.UpdatePolicy {
 	return m.Policy
 }
 
+// CompositeDeleter is a mock that implements CompositeDeleter interface.
+type CompositeDeleter struct{ Policy *xpv1.CompositeDeletePolicy }
```

Comment:
Just thinking out loud here, but I'm wary of the amount of "machinery fields" we're injecting into claims and XRs - fields that pertain to Crossplane rather than to the API abstractions folks are trying to build. A little part of me wishes we'd either homed them all under a parent field (e.g. `spec.crossplane`) or leaned more into annotations.

Unfortunately I think that ship has sailed, and thus your approach of adding this new policy as a top-level spec field is the most idiomatic and consistent one I can think of.